File: app/src/main/java/ru/garretech/garred/doramatv/database/MovieDAO.java
package ru.garretech.garred.doramatv.database;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import ru.garretech.garred.doramatv.model.Movie;

@Dao
public interface MovieDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long addMovie(Movie movie);


    @Query("SELECT * FROM movie")
    List<Movie> getAllCachedMovies();

    @Query("SELECT * FROM movie WHERE URL = :URL")
    Movie getMovie(String URL);

}

File: app/src/main/java/ru/garretech/garred/doramatv/database/MovieDAO.kt
package ru.garretech.garred.doramatv.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

import ru.garretech.garred.doramatv.model.Movie

@Dao
interface MovieDAO {


    @get:Query("SELECT * FROM movie")
    val allCachedMovies: List<Movie>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addMovie(movie: Movie): Long

    @Query("SELECT * FROM movie WHERE URL = :URL")
    fun getMovie(URL: String): Movie?

}
